src/Console/Styling/RenderCommandNotFound.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Styling;
+
+use Tempest\Console\ConsoleCommand;
+use Tempest\Console\ConsoleConfig;
+use Tempest\Console\RenderConsoleCommand;
+
+final readonly class RenderCommandNotFound
+{
+    public function __construct(
+        protected ConsoleConfig $consoleConfig,
+        protected RenderConsoleCommand $renderConsoleCommand,
+    ) {
+
+    }
+
+    public function __invoke(string $command): string
+    {
+        $similarCommands = $this->getSimilarCommands($command);
+
+        return ConsoleOutputBuilder::new()
+            ->error("Command `{$command}` not found")
+            ->blank()
+            ->when(! ! $similarCommands, function (ConsoleOutputBuilder $response) use ($similarCommands) {
+                $response->warning('Did you mean one of these commands?')
+                    ->blank();
+
```

Comment:
Something we could add along the way is to allow the user to pick the command automatically from the error output without having to restart the console session. But doesn't need to be added now.